plugins/modules/sonic_ssh_server.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2025 Dell Inc. or its subsidiaries. All Rights Reserved.
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+The module file for sonic_ssh_server
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+module: sonic_ssh_server
+version_added: '3.1.0'
+short_description: Manage SSH server configurations on SONiC
+description:
+  - This module provides SSH server configuration management to specify the
+     algorithms used for SSH connection in devices running SONiC.
+author: Bhavesh (@bhaveshdell)
+options:
+  config:
+    description:
+      - SSH servers use the following configurations for SSH connections.
```

Comment:
Please add a note here or for the "server_globals" description section stating that deletion of any of the "str" type options results in setting the configuration for the "deleted" option to the set of default values.